src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/UniversalCryptoOneShot.cs

```diff
@@ -20,6 +20,10 @@ internal static class UniversalCryptoOneShot
             if (input.Length % cipher.PaddingSizeInBytes != 0)
                 throw new CryptographicException(SR.Cryptography_PartialBlock);
 
+            bool cipherHandlesPadding = cipher.HandlesPadding;
+            if (cipherHandlesPadding)
```

Comment:
I think most of the changes in this method can be undone based on how the implementation works. I think you can just handle `cipherHandlesPadding` up-front:

```C#
if (cipherHandlesPadding)
{
    cipher.ValidatePaddingMode(paddingMode);
    bytesWritten = cipher.TransformFinal(input, output);
    return true;
}

// The rest of the method
```

The rest of this decryption implementation is about handling various padding scenarios, we we don't have them here since the browser implementation does padding for us.